src/main/java/org/prebid/server/auction/BidResponseCreator.java

```diff
@@ -267,40 +261,20 @@ private String updateBidAdm(
                 : bidAdm;
     }
 
-    // will be updated in https://github.com/prebid/prebid-server-java/pull/1126
-    private ObjectNode updateBidExt(
-            Bid bid,
-            BidType bidType,
-            String bidder,
-            Account account,
-            VideoStoredDataResult videoStoredDataResult,
-            EventsContext eventsContext,
-            String generatedBidId,
-            String effectiveBidId) {
+    private ObjectNode updateBidExt(Bid bid, BidType bidType, String bidder, Account account,
+            VideoStoredDataResult videoStoredDataResult, EventsContext eventsContext,
+            String generatedBidId, String effectiveBidId) {
 
         final ExtBidPrebid updatedExtBidPrebid = updateBidExtPrebid(
                 bid, bidType, bidder, account, videoStoredDataResult, eventsContext, generatedBidId, effectiveBidId);
-
         final ObjectNode existingBidExt = bid.getExt();
-        JsonNode skadnObject = mapper.mapper().createObjectNode();
-        JsonNode origBidPrice = null;
-        JsonNode origBidCur = null;
+
+        final ObjectNode updatedBidExt = mapper.mapper().createObjectNode();
+        updatedBidExt.set(PREBID_EXT, mapper.mapper().valueToTree(updatedExtBidPrebid));
+
         if (existingBidExt != null && !existingBidExt.isEmpty()) {
-            skadnObject = getAndRemoveProperty(SKADN_PROPERTY, existingBidExt);
-            origBidPrice = getAndRemoveProperty(ORIGINAL_BID_CPM, existingBidExt);
-            origBidCur = getAndRemoveProperty(ORIGINAL_BID_CURRENCY, existingBidExt);
-        }
-        final ObjectNode extPrebidBidder = existingBidExt != null && !existingBidExt.isEmpty() ? existingBidExt : null;
-        final ExtPrebid<ExtBidPrebid, ObjectNode> bidExt = ExtPrebid.of(updatedExtBidPrebid, extPrebidBidder);
-        final ObjectNode updatedBidExt = mapper.mapper().valueToTree(bidExt);
-        if (skadnObject != null && !skadnObject.isEmpty()) {
-            updatedBidExt.set(SKADN_PROPERTY, skadnObject);
-        }
-        if (origBidPrice != null) {
-            updatedBidExt.set(ORIGINAL_BID_CPM, origBidPrice);
-        }
-        if (origBidCur != null) {
-            updatedBidExt.set(ORIGINAL_BID_CURRENCY, origBidCur);
+            existingBidExt.remove(PREBID_EXT);
+            updatedBidExt.setAll(existingBidExt);
```

Comment:
Seems technically we can rewrite updated fields from `updatedExtBidPrebid`.
That's why you've doing `remove(PREBID_EXT);`.

What if we'll set `updatedExtBidPrebid` after creating `updatedBidExt` from `existingBidExt`.
So, we don't need removing then, right?